src/orders/components/OrderCustomerAddressesEditDialog/utils.ts

```diff
@@ -48,6 +49,21 @@ export function validateDefaultAddress<T extends AddressFragment>(
   return defaultAddress;
 }
 
+const filterAddressErrors = (
+  dialogErrors: Array<OrderErrorFragment | AccountErrorFragment>,
+  addressType: AddressTypeEnum
+) => dialogErrors.filter(error => error.addressType === addressType);
+
+interface ShippingAddresses {
+  shippingAddress: AccountErrorFragment[] | AddressInput;
+  billingAddress: AccountErrorFragment[] | AddressInput;
+}
+
+export const hasPreSubmitErrors = (input: ShippingAddresses) =>
+  Object.values(input)
+    .flat()
+    .some(el => (el as AccountErrorFragment).code);
```

Comment:
It looks like you're having to type cast because `AddressInput ` does not have a `code` property and you might be getting the `property does not exist on type ...`. To solve this you can use a type safe check instead:
```suggestion
    .some(el => "code" in el);
```

This is also a natively built type guard (automatic type assertion in union types), here's a short example on how it can be useful:
```
type A = { foo: string, ok: boolean };
type B = { bar: string  }

const someFn = (x: A | B) => {
  if ("foo" in x) {
    return x.ok; // does not throw type error
  }
}
```